internal/rbd/nodeserver.go

```diff
@@ -1042,8 +1065,18 @@ func (ns *NodeServer) NodeExpandVolume(
 	volumePath := req.GetStagingTargetPath()
 	if volumePath == "" {
 		// If Kubernetes version < v1.19.0 the volume_path would be
-		// having the staging_target_path information
-		volumePath = req.GetVolumePath()
+		// having the staging_target_path information.
+		// stagingTargetPath is optional parameter in NodeExpandVolumeRequest request
+		// Kubernetes will not send the staging_target_path in the volume_path in some cases
+		// Refer https://github.com/kubernetes/kubernetes/issues/115343
+		var err error
+		volumePath, err = ns.getStagingPath(volumePath)
+		if err != nil {
+			// If stagingTargetPath is not found in volumePath then use
+			// volumePath
+			volumePath = req.GetVolumePath()
+			log.UsefulLog(ctx, "failed to find stagingTargetPath from volumePath %v: %v", volumePath, err)
+		}
```

Comment:
@Madhu-1 
Did you intend the code to the as follows ? 

Otherwise the `volumePath` being sent to `getStagingPath()` is empty from the check on line 1066

```suggestion
		var err error
		volumePath, err = ns.getStagingPath(req.GetVolumePath())
		if err != nil {
			// If stagingTargetPath is not found in volumePath then use
			// volumePath
			volumePath = req.GetVolumePath()
			log.UsefulLog(ctx, "failed to find stagingTargetPath from volumePath %v: %v", volumePath, err)
		}
```

Reply:
Yes that correct, fixed now.